src/Tests/dotnet.Tests/dotnet-workload-clean/GivenDotnetWorkloadClean.cs

```diff
@@ -0,0 +1,199 @@
+
+        private (string testDirectory, string dotnetRoot, string userProfileDir, WorkloadResolver workloadResolver, MockNuGetPackageDownloader nugetDownloader) Setup(bool userLocal, bool cleanAll)
+        {
+            var testDirectory = _testAssetsManager.CreateTestDirectory(identifier: userLocal ? $"userlocal-{cleanAll}" : $"default-{cleanAll}").Path;
+            var dotnetRoot = Path.Combine(testDirectory, dotnet);
+            var userProfileDir = Path.Combine(testDirectory, _profileDirectoryLeafName);
+            var workloadResolver = WorkloadResolver.CreateForTests(new MockManifestProvider(new[] { _manifestPath }), dotnetRoot, userLocal, userProfileDir);
+            var nugetDownloader = new MockNuGetPackageDownloader(dotnetRoot);
+
+            return (testDirectory, dotnetRoot, userProfileDir, workloadResolver, nugetDownloader);
+        }
+
+        public GivenDotnetWorkloadClean(ITestOutputHelper log) : base(log)
+        {
+            _reporter = new BufferedReporter();
+            _manifestPath = Path.Combine(_testAssetsManager.GetAndValidateTestProjectDirectory("SampleManifest"), "Sample.json");
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(false, false)]
+        public void GivenWorkloadCleanFileBasedItRemovesPacksAndPackRecords(bool userLocal, bool cleanAll)
+        {
+            var (testDirectory, dotnetRoot, userProfileDir, workloadResolver, nugetDownloader) = Setup(userLocal, cleanAll);
+
+            string installRoot = userLocal ? userProfileDir : dotnetRoot;
+            if (userLocal)
+            {
+                WorkloadFileBasedInstall.SetUserLocal(dotnetRoot, _sdkFeatureVersion);
+            }
+
+            // Test
+            InstallWorkload(userProfileDir, dotnetRoot, testDirectory, workloadResolver, nugetDownloader);
+
+            var extraPackRecordPath = MakePackRecord(installRoot);
+            var extraPackPath = MakePack(installRoot);
+
+
+            var cleanCommand = cleanAll ? GenerateWorkloadCleanAllCommand(workloadResolver, userProfileDir, dotnetRoot) : GenerateWorkloadCleanCommand(workloadResolver, userProfileDir, dotnetRoot);
+            cleanCommand.Execute();
+
+            AssertExtraneousPacksAreRemoved(extraPackPath, extraPackRecordPath);
+            AssertValidPackCountsMatchExpected(installRoot, expectedPackCount: cleanAll ? 0 : 7, expectedPackRecordCount: cleanAll ? 0 : 8);
+
+            AssertAdjacentCommandsStillPass(userProfileDir, dotnetRoot, testDirectory, workloadResolver, nugetDownloader);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void GivenWorkloadCleanAllFileBasedItCleansAllFeatureBands(bool userLocal)
+        {
+            var (testDirectory, dotnetRoot, userProfileDir, workloadResolver, nugetDownloader) = Setup(userLocal, true);
```

Comment:
Nit for future reference (I don't suggest changing this here, especially since it's in a dead-end release branch):  It can be good for readability to specify the parameter names when passing in literals to a method.


```suggestion
            var (testDirectory, dotnetRoot, userProfileDir, workloadResolver, nugetDownloader) = Setup(userLocal, cleanAll: true);
```

Reply:
Good to know, thanks :)